### DISCUSSION TITLE
K8S native file cache support

### DISCUSSION THREAD

**Discussion by dishkakrauch** [trinodb/trino#24395] [General]
Hello there.

Do you have any plans to add helm chart StatefulSet support for the following guid https://trino.io/docs/current/object-storage/file-system-cache.html? Looks like it's best soluiton for baremetal installation but not for the K8S with Deployemnts. There are workarounds:

- use emptyDir ram mount
- handly addind additional disk to K8S node
- switch to S3 storage (depercated and not gonna make cache works better)

**Comment by dishkakrauch** [comment]
Any thoughts?

**Comment by hashhar** [comment]
the cache doesn't need to be stateful at all in the first place.

Also statefulsets are usually backed by network attached storage instead of direct attached storage which is slow and the latency removes all the performance benefit of using a local cache in the first place.

The suggestion is to point the cache directory to a local NVME disk which isn't managed by k8s itself (there's no need to do that).

Feel free to discuss with the helm chart maintainers by creating issue at the https://github.com/trinodb/charts repo.
